### PR TITLE
Add reactive-lb support for WebClient

### PR DESCRIPTION
**Pull request by nkolosnjaji**

### CODE REVIEW

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package org.springframework.cloud.loadbalancer.core;
```
Comment (ryanjbaxter) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java]: Needs license header

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * @author Spencer Gibb
```
Comment (ryanjbaxter) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java]: Please add your own name as the author

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java [hunk 3]
```diff
@@ -50,7 +51,8 @@ public LoadBalancerClientFactory loadBalancerClientFactory() {
 	}
 	
 	@Bean
-	public ReactiveLoadBalancerExchangeFilterFunction loadBalancerExchangeFilterFunction() {
+	@ConditionalOnBean(LoadBalancerClientSpecification.class)
```
Comment (spencergibb) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java]: That's incorrect. See https://github.com/spring-cloud/spring-cloud-commons/blob/60d2a1ead3f5a0c4f5347978a60d2206453080e5/spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/ReactiveLoadBalancerAutoConfiguration.java#L22 for what to do.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java (hunk 2 above)
Comment (nkolosnjaji) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java]: done

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java (hunk 1 above)
Comment (nkolosnjaji) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/ReactiveLoadBalancerExchangeFilterFunction.java]: done

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java (hunk 3 above)
Comment (nkolosnjaji) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java]: done

**Comment by nkolosnjaji** [comment]
Sure, I guess you mean `@LoadBalanced` annotation :)

**Comment by nkolosnjaji** [comment]
@spencergibb anything else needs to be done?

**Comment by nkolosnjaji** [comment]
@spencergibb should I rebase?

**Comment by spencergibb** [comment]
yes please

**Comment by nkolosnjaji** [comment]
done

**Comment by ryanjbaxter** [comment]
We probably now want to submit this against the 2.0.x branch and we will merge it forward.  Also documentation around the support for this would be useful.

**Comment by nkolosnjaji** [comment]
Sure, I will try to find some time this week.

**Comment by nkolosnjaji** [comment]
@ryanjbaxter I'm not sure if this kind of details are needed in documentation and what will happen with other sections (like 2.4 & 2.6). 
I wanted to give an example which will be similar to _Ribbon_ + _Eureka_ setup.

P.S. Sorry, I forgot to add [skip ci] to commit message.

**Comment by ryanjbaxter** [comment]
@nkolosnjaji can you clean up this PR?

**Comment by nkolosnjaji** [comment]
Sure, but branch 2.0.x does not contain spring-cloud-loadbalancer module.
@ryanjbaxter Did you mean 2.1.x instead?

**Comment by ryanjbaxter** [comment]
Yes

**Comment by nkolosnjaji** [comment]
Done

**Comment by ryanjbaxter** [comment]
I think this is also going to suffer from https://github.com/spring-cloud/spring-cloud-commons/issues/557

**Comment by nkolosnjaji** [comment]
Ok, I can make changes once reactive discovery client is in place.

**Comment by OlgaMaciaszek** [comment]
Hello @nkolosnjaji Thank you for your time that you've spent working on this PR. Currently, we have started working on some broader changes involving `spring-cloud-loadbalancer`. The reactive `@LoadBalanced WebClient` is being addressed by [this issue](https://github.com/spring-cloud/spring-cloud-commons/issues/576) and the corresponding PR, so will be closing this one.

**Pull request closed without merging**